### ISSUE TITLE
The SSL/TLS function of your client is deactivated or the certificate can't be read

### ISSUE THREAD

**Issue by MaxFoton** [eiskaltdcpp/eiskaltdcpp#462]
how can I do all this? I didn't find any certificates for the EiskaltDC program.

	Please do the following:

	 - Go in your client settings in the menu "Security Certs"
	 - Check if the path for the private key is correct. (YourClient\Certificates\client.key)
	 - Check if the path for the cert is correct. (YourClient\Certificates\client.crt)
	 - And if the cert is really in that directory. (YourClient\Certificates\)
	 - Then set all marks at the TLS features.

	If it doesn't work it eventually helps to generate a new cert. To make that go:

	 - into your client settings in the menu "Security Certs"
	 - Delete the whole path in all 3 boxes and then click OK.
	 - Close your client.
	 - Delete these files: Certificates/client.crt & Certificates/client.key
	 - Start client and then click in the settings/security menu the "generate" button.
	 - Restart your client.

**Comment by tehnick** [comment]
Which version of program do you use? In which operation system?

**Comment by MaxFoton** [comment]
version 2.2.10 Ubuntu 20.10

**Comment by tehnick** [comment]
We have official PPA for Ubuntu. The latest stable version is `2.4.1` and I strongly suggest to update the program.

As for managing of security certs, currently only GTK+ UI has special section in settings dialog for this. In Qt UI it will be added in one of the next versions.

Have in mind that GTK+ UI is bogus and is not recommended for permanent usage. It will be removed in next major release of program. But you may launch GTK+ UI one time for changing security certs related settings and then use Qt UI on regular basis.

Another options is managing of keys in file manager (see: `~/.config/eiskaltdc++/Certificates/`) and any text editor (see: `~/.config/eiskaltdc++/DCPlusPlus.xml`).

Do not forget to exit from Qt UI before launching of GTK+ UI or before editing of `DCPlusPlus.xml` in a text editor.

**Comment by MaxFoton** [comment]
tnx. but version 2.4.1 is not stable, and designed for the Hairy Hippopotamus) I have Groovy Ubuntu

**Comment by MaxFoton** [comment]
"It looks that you have no idea what the PPA is and how to use it in Ubuntu..."  yes, I did not understand at all. and I didn't find any detailed instructions on github.

**Comment by tehnick** [comment]
Back to the main question: have you configured security certs?

**Comment by MaxFoton** [comment]
no. this is too difficult for me( I can't update the program, I added it from the repository, the packages were updated, but the program version is not. do you speak russian

**Comment by tehnick** [comment]
> I can't update the program

Official PPA is [here](https://launchpad.net/~tehnick/+archive/ubuntu/direct-connect).

Just few commands in terminal:
```
sudo add-apt-repository ppa:tehnick/direct-connect
sudo apt update
sudo apt upgrade -V
```
And you will have latest stable version of program.

> do you speak russian

Yep.

**Comment by MaxFoton** [comment]
спасибо! программа обновилась. теперь не понимаюю как увидеть инфо когда набираешь в хабе команду пользователя, в AirDC++ для винды все работало

**Comment by tehnick** [comment]
> как увидеть инфо когда набираешь в хабе команду пользователя

Ничего не понял.

Команды пользователя задаются один раз в настройках, после чего они доступны в контекстном меню хабов вместе с командами, которые присылает сам хаб.

> в AirDC++ для винды все работало

Это вообще ни разу не информативно. Если нужно все как в AirDC++, то никто не мешает использовать виндовую версию AirDC++ из WINE. Если просто что-то не получается найти, то сформулируйте вопрос нормально. Кстати, в баг-трекере есть возможность прикреплять скриншоты...

А еще существуют нативные версии AirDC++ Web UI для линукса. Но они на любителя, конечно.

Ну и на вопрос про настройку сертификатов безопасности ответьте. Разобрались с настройкой или нет?

**Comment by MaxFoton** [comment]
он мне теперь такую ошибку выдает Ошибка: Ошибка SSL: error:00000005:lib(0):func(0):DH lib (-1, 5)...
команды пользователя сами заработали. мне Eiskalt больше нравится чем AirDC

**Comment by tehnick** [comment]
Хаб публичный? Какой у него адрес?
